src/ChannelServer/World/Character.cs

```diff
@@ -555,6 +639,392 @@ public int MaxMAtk
+				//	rate += 32 + ((SkillLevel-1) * 2)
+				var rate = 0;
+				var byRateBuffs = (float)Math.Floor(value * rate);
+
+				value += byBuffs + byRateBuffs;
+
+				return (int)value;
+			}
+		}
+
+		/// <summary>
+		/// Returns Critical Resistance.
+		/// </summary>
+		public int CritRes
+		{
+			get
+			{
+				var Level = this.Level;
+
+				var byLevel = Level / 2f;
+				var byItem = 0; // TODO: Cached Crit Resistance for inventory/equip
+
+				var value = byLevel + byItem;
+
+				// Buffs: "CRTDR_BM"
+				var byBuffs = 0;
+
+				// Rate buffs:
+				var rate = 0;
+				var byRateBuffs = (float)Math.Floor(value * rate);
+
+				value += byBuffs + byRateBuffs;
+
+				return (int)value;
+			}
+		}
+
+		/// <summary>
+		/// Returns Accuracy.
+		/// </summary>
+		public int Accuracy
+		{
+			get
+			{
+				var Level = this.Level;
+				var stat = this.Str;
+
+				var byLevel = Level / 4f;
+				var byStat = (stat / 2f) + ((float)Math.Floor(stat / 15f) * 3f);
+				var byItem = 0; // TODO: Cached Accuracy for inventory/equip
+
+				var value = byLevel + byStat + byItem;
+
+				// Buffs: "HR_BM"
+				var byBuffs = 0;
+
+				// Rate buffs: Does tos have something like Accuracy + x% ?
+				var rate = 0;
+				var byRateBuffs = (float)Math.Floor(value * rate);
+
+				value += byBuffs + byRateBuffs;
+
+				return (int)value;
+			}
+		}
+
+		/// <summary>
+		/// Returns Evasion.
+		/// </summary>
+		public int Evasion
+		{
+			get
+			{
+				var Level = this.Level;
+				var stat = this.Dex;
+
+				var byLevel = Level / 4f;
+				var byStat = (stat / 2f) + ((float)Math.Floor(stat / 15f) * 3f);
+				var byItem = 0; // TODO: Cached Evasion for inventory/equip
+
+				var value = byLevel + byStat + byItem;
+
+				//if(hasBuff("Finestra"))
+				//	if(hasAttribute("Finestra: Splash")) | reduces evasion rate by x2 more of the existing value when [Finestra] is active
+				//		value -= (15 + ((SkillLevel - 1) * 15)) * 2
+				//	else
+				//		value -= 15 + ((SkillLevel - 1) * 15)
+
+
+
+				// Buffs: "DR_BM"
+				var byBuffs = 0;
+
+				// Rate buffs: Can't find the right property
+				//if(hasBuff("Evasive Action"))
+				//	rate += SkillLevel*2
+				var rate = 0;
+				var byRateBuffs = (float)Math.Floor(value * rate);
+
+				value += byBuffs + byRateBuffs;
+
+				return (int)value;
+			}
+		}
+
+		/// <summary>
+		/// Returns Block.
+		/// </summary>
+		public int Block
+		{
+			get
+			{
+				var byItem = 0f; // TODO: Cached Block for inventory/equip ("BLK")
+
+				var value = byItem;
+
+				//if(hasBuff("High Guard"))
+				//	value += 100 + ((SkillLevel - 1) * 30)
+				//if(hasBuff("Finestra"))
+				//	value += 40 + ((SkillLevel - 1) * 15)
+
+				// Buffs: "BLK_BM"
+				var byBuffs = 0;
+
+				// Rate buffs:
+				var rate = 0;
+				var byRateBuffs = (float)Math.Floor(value * rate);
+
+				value += byBuffs + byRateBuffs;
+
+				return (int)value;
+			}
+		}
+
+		/// <summary>
+		/// Returns Block Penetration.
+		/// </summary>
+		public int BlockPen
```

Comment:
Feel free to use more descriptive names, like "Defense" and "BlockPenetration". No reason to cut corners, IntelliSense will complete it anyway^^

Reply:
I wasn't sure about that, I'll change it